test/apps/iodemo/io_demo.cc

```diff
@@ -282,15 +289,33 @@ class P2pDemoCommon : public UcxContext {
             return tr_hdr()->sn;
         }
 
-        inline uint16_t calc_chksum() const {
-            uint16_t chk_sum = ucs_crc16(tr_hdr(),
-                                         _iov[0]->size() - sizeof(chk_hdr_t));
-
-            for (size_t i = 1; i < _iov.size(); ++i) {
-                chk_sum ^= ucs_crc16(_iov[i]->buffer(), _iov[i]->size());
+        iovec_arr_t to_iovec_arr() {
```

Comment:
IMO it's not very nice to convert BufferIov to vector<iovec> just to handle io messages
other options:
1. add method to construct a (temporary) BufferIov from io_msg (void*,size_t)
2. make validate() template function on so it can handle both cases

Reply:
1. io_msg -> BufferIov  would require extra memcpy since BufferIov impl is based on Buffer from MemPool
2. validate() template function would require specialization for every type since iovec and BufferIov have diff APIs, so it becomes the same as 2 diff validate() funcs

Reply:
maybe io_msg could be based on Buffer class as well?

Reply:
recv io_msg is a part of UcxContext